### PR TITLE
feat: added Apple auth provider

### PR DESCRIPTION
**Pull request by EloB**
### What does it do?

I've added Apple auth provider.

The biggest change is in `packages/plugins/users-permissions/admin/src/pages/Providers/utils/forms.js`. It might look like a lot of changes but mostly extraction of code into `commonFields` and `commonSchema` that I reuse to create each form layout/schema.

### Why is it needed?

Apple has a policy that you NEED to include Apple ID in mobile app if you include other OAuth providers. iOS is very important for App development. To make Strapi more competitive alternative on the market I think this should be included.
https://developer.apple.com/app-store/review/guidelines/#sign-in-with-apple

### How to test it?

Works like any other OAuth provider. Visit https://yourstrapi.com/api/connect/apple.

### Related issue(s)/PR(s)

Let us know if this is related to any issue/pull request
https://github.com/strapi/rfcs/pull/20

Screenshot 2023-01-23 at 02 06 40

My first Strapi PR :)

### CODE REVIEW

File: packages/plugins/users-permissions/admin/src/components/FormModal/index.js [hunk 1]
```diff
@@ -66,7 +66,9 @@ const FormModal = ({
                               {...input}
                               error={errors[input.name]}
                               onChange={handleChange}
-                              value={values[input.name]}
+                              value={
+                                input.name === 'noName' ? values.redirectUri : values[input.name]
```
Comment (EloB) [inline packages/plugins/users-permissions/admin/src/components/FormModal/index.js:70]: I've also fixed the textfield with "The redirect URL to add in your PROVIDER application configurations". So it shows the correct value from backend.

**Comment by EloB** [comment]
I will also control that it works with external login (in-app login). Didn’t have time to control that yesterday.

The web auth was working for me but their OAuth solution is really weird. A second eye is very important.

**Comment by Marc-Roig** [comment]
Hello! we appreciate the work and the proposal for this new provider.
Unfortunately , as to properly test it we have to purchase a 3rd party licence, we decided in Strapi that we wouldn't maintain providers (or anything) that requires us to maintain such a license to test it.

We are refactoring the user-permissions plugin, and we might add the possibility for users to add their own custom providers. That is where this code could fit in the future.

Unfortunately now we can not accept this work :( .

**Comment by EloB** [comment]
@Marc-Roig I've sent a mail to Apple developer support and ask them if there is anything they can do about this.

**Comment by Marc-Roig** [comment]
🙌 Thank you @EloB

**Pull request closed without merging**